### PR TITLE
fixing problems

### PR DESCRIPTION
**Pull request by mouhamamd-kl**
this change the problems in the Delegate when we assign the test to it
so we add Action to the test method
also the Chunck method which do not work in the old version of vs so added a new method to do it as exstintion to the array and when using Combine and remove also added the Action

### CODE REVIEW

File: AdvancedProgramming2.csproj [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net7.0</TargetFramework>
+    <TargetFramework>net48</TargetFramework>
```
Comment (ma7m00dma7m00d) [inline AdvancedProgramming2.csproj:5]: Just download and install dotnet 7

File: Program.cs [hunk 2]
```diff
@@ -1,5 +1,24 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
 namespace Delegates
 {
+    public static class ArrayExtensions
```
Comment (ma7m00dma7m00d) [inline Program.cs:8]: An excellent method, but it already exists in System.Collections.Generic

Comment (mouhamamd-kl) [reply Program.cs:8]: not in the older versions of vs like 2019 and before

File: AdvancedProgramming2.csproj (hunk 1 above)
Comment (mouhamamd-kl) [reply AdvancedProgramming2.csproj:5]: i did download it but i made the pull request so other student won't have to download any thing and just clone the repo

**Pull request closed without merging**